File: Kotlin/bee 1018.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main(args: Array<String>) {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val valor = br.readLine().toInt()

    val notas100 = valor/100;
    var resto = valor % 100

    val notas50 = resto/50
    resto %= 50

    val notas20 = resto/20
    resto %= 20

    val notas10 = resto/10
    resto %= 10

    val notas5 = resto/5
    resto %= 5

    val  notas2 = resto/2
    resto %= 2

    val notas1 = resto

    println(valor)
    println(String.format("$notas100 nota(s) de R$ 100,00\n$notas50 nota(s) de R$ 50,00\n$notas20 nota(s) de R$ 20,00\n$notas10 nota(s) de R$ 10,00\n$notas5 nota(s) de R$ 5,00\n$notas2 nota(s) de R$ 2,00\n$notas1 nota(s) de R$ 1,00"))

}

File: Rust/bee_1018.rs
use std::io::{self, Read};

fn main() {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).unwrap();

    let valor: i32 = input.trim().parse().unwrap();

    let notas_100 = valor / 100;
    let mut resto = valor % 100;

    let notas_50 = resto / 50;
    resto %= 50;

    let notas_20 = resto / 20;
    resto %= 20;

    let notas_10 = resto / 10;
    resto %= 10;

    let notas_5 = resto / 5;
    resto %= 5;

    let notas_2 = resto / 2;
    resto %= 2;

    let notas_1 = resto;

    println!("{}", valor);
    println!("{} nota(s) de R$ 100,00", notas_100);
    println!("{} nota(s) de R$ 50,00", notas_50);
    println!("{} nota(s) de R$ 20,00", notas_20);
    println!("{} nota(s) de R$ 10,00", notas_10);
    println!("{} nota(s) de R$ 5,00", notas_5);
    println!("{} nota(s) de R$ 2,00", notas_2);
    println!("{} nota(s) de R$ 1,00", notas_1);
}
